src/lambda/explorer/cloudFormationNode.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import CloudFormation = require('aws-sdk/clients/cloudformation')
+
+import { TreeItem, TreeItemCollapsibleState, Uri } from 'vscode'
+
+import { ext } from '../../shared/extensionGlobals'
+import { AWSTreeNodeBase } from '../../shared/treeview/awsTreeNodeBase'
+import { FunctionNode } from './functionNode'
+import { NoFunctionsNode } from './noFunctionsNode'
+
+export class CloudFormationNode extends AWSTreeNodeBase implements TreeItem {
+    public static contextValue: string = 'awsCloudFormation'
+    public contextValue: string = CloudFormationNode.contextValue
+
+    public label?: string
+    public tooltip?: string
+    public iconPath?: { light: Uri; dark: Uri }
+
+    public readonly lambdaResources: string[] = []
+    public stackDescribed: boolean = false
```

Comment:
Is this public for testing purposes only? If so I recommend making it protected, and using a derived test class to work with it in tests.

Reply:
Yes, they were public for testing purposes and can be made private once we start using mocking techniques. I use the stackDescribed flag to avoid querying the stack resources every time the method getChildren gets called (but I don't know if this makes sense). I created a derived test class as suggested.

Reply:
Sounds good - I agree we don't want to re-query each time getChildren is called.